nvtabular/ops/target_encoding.py

```diff
@@ -166,6 +166,9 @@ def fit(self, col_selector: ColumnSelector, ddf: dd.DataFrame):
         moments = None
         if self.target_mean is None:
             # calcualte the mean if we don't have it already
+            self.target = (
```

Comment:
> So yeah... went ahead and tried out your way and it actually fails (you can look at failed test runs). The reason this doesn't work is because in the constructor your target might be a node and you wont have the target columns until you present the dataset. This is why it has to be done in the fit stage. You could put it in the fit schema but that does not seem correct because we run that multiple times. And we only run the fit once. In any given cycle.

ok, that makes sense